clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -4945,8 +4945,7 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
   case Builtin::BIprintf:
     if (getTarget().getTriple().isNVPTX())
       return EmitNVPTXDevicePrintfCallExpr(E, ReturnValue);
-    if (getTarget().getTriple().getArch() == Triple::amdgcn &&
-        getLangOpts().HIP)
+    if (getTarget().getTriple().getArch() == Triple::amdgcn)
```

Comment:
I think changes like this should be committed to llorg directly.
There is SYCL language mode support in llorg, which should be enough to justify and validate such changes.
Committing to llorg will significantly help us to reduce the downstream maintenance overhead.

Please, consider extracting the part that do not depend on DPC++ changes, which are not upstreamed to llorg yet and commit this part directly to llorg.
We are syncing with llorg on weekly basis, so if changes merged to llvm-project, they expected to appear in DPC++ within a week.

Reply:
This code does not seem to be completely needed and can be removed.